src/Microsoft.AspNet.Razor/Parser/CSharpCodeParser.Directives.cs

```diff
@@ -320,6 +327,34 @@ private void TagHelperDirective(string keyword, ISpanChunkGenerator chunkGenerat
                 // Parse to the end of the line. Essentially accepts anything until end of line, comments, invalid code
                 // etc.
                 AcceptUntil(CSharpSymbolType.NewLine);
+
+                // Pull out the value and remove whitespaces and optional quotes
+                var rawValue = Span.GetContent().Value.Trim();
+
+                var startsWithQuote = rawValue.StartsWith("\"", StringComparison.Ordinal);
+                var endsWithQuote = rawValue.EndsWith("\"", StringComparison.Ordinal);
+                if (startsWithQuote != endsWithQuote)
+                {
+                    Context.OnError(
+                        startLocation,
+                        RazorResources.FormatParseError_DirectiveMustBeSurroundedByQuotes(keyword),
```

Comment:
Need to update the error to specify the optionality of the quotes. Currently it sounds like a requirement.